lib/src/model.dart

```diff
@@ -3215,6 +3216,15 @@ abstract class ModelElement extends Canonicalization
     return _documentationFrom;
   }
 
+  bool get hasSourceHref => sourceHref != '';
+  String _sourceHref;
+  String get sourceHref {
+    if (_sourceHref == null) {
```

Comment:
`_sourceHref ??= new SourceLinker.fromElement(this).href();` ?

Reply:
done